### PR TITLE
Web auth loading

### PR DESCRIPTION
**Pull request by showduhtung**
- Implemented basic loading logic similar to the mobile app
- created loading modals for any routes that require them
- authenticate login
- clears out on logout
- has a remember me function
- has a routed path for to reset token
- if reset token is invalid, sends auth email
- auth email is the same component as forgot password
- all passwords on forms are hidden

### CODE REVIEW

File: src/components/shared/TransitionsModal.tsx [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+    open: boolean;
+    setOpen: (arg0: boolean) => void;
+}
+
+export const TransitionsModal = ({ open, setOpen }: TransitionsModalProps) => {
+    const classes = useStyles();
+    const status = useSelector(({ load }) => load.loadStatus.AUTH);
+
+    return (
+        <div>
```
Comment (whoahs) [inline src/components/shared/TransitionsModal.tsx]: is this div necessary?

File: src/components/shared/AuthEmail.tsx [hunk 2]
```diff
@@ -30,28 +34,50 @@ const useStyles = makeStyles((theme) => ({
...
             <div className={classes.paper}>
                 <Typography component="h1" variant="h5">
-                    Forgot Password
-                </Typography>
-                <Typography>
-                    Lost your password? Please enter your email address. You will receive
-                    a link to create a new password via email.
+                    {data.title}
                 </Typography>
+                {data.description ? <Typography>{data.description} </Typography> : null}
```
Comment (whoahs) [inline src/components/shared/AuthEmail.tsx]: instead of ternaries using null for all of these, can just do `{someBoolean && <MyComponent />}`

**Review by whoahs — APPROVED** [review]

File: src/components/shared/TransitionsModal.tsx (hunk 1 above)
Comment (showduhtung) [reply src/components/shared/TransitionsModal.tsx]: good catch

File: src/components/shared/AuthEmail.tsx (hunk 2 above)
Comment (showduhtung) [reply src/components/shared/AuthEmail.tsx]: ohh TIL

**Pull request merged**